vllm/v1/request.py

```diff
@@ -134,18 +149,70 @@ def get_num_encoder_tokens(self, input_id: int) -> int:
         num_tokens = self.mm_positions[input_id]["length"]
         return num_tokens
 
+    @property
+    def use_guided_decoding(self) -> bool:
+        return self.sampling_params.guided_decoding is not None
+
+    @functools.cached_property
+    def guided_decoding_key(self) -> GuidedDecodingKey:
+        params = self.sampling_params.guided_decoding
+        assert params is not None, "params can't be None."
+        if params.json is not None:
+            if not isinstance(params.json, str):
+                json_str = json.dumps(params.json)
+            else:
+                json_str = params.json
+            return (GuidedDecodingOptions.json, json_str)
+        elif params.json_object:
+            return (GuidedDecodingOptions.json_object, "")
+        elif params.regex is not None:
+            return (GuidedDecodingOptions.regex, params.regex)
+        elif params.choice is not None:
+            if not isinstance(params.choice, str):
+                json_str = json.dumps(params.choice)
+            else:
+                json_str = params.choice
+            return (GuidedDecodingOptions.choice, json_str)
+        elif params.grammar is not None:
+            return (GuidedDecodingOptions.grammar, params.grammar)
+        else:
+            raise ValueError("No valid guided decoding parameter found")
+
+    def _check_grammar_completion(self) -> bool:
+        if isinstance(self._grammar, Future):
+            try:
+                self._grammar = self._grammar.result(timeout=0.0001)
```

Comment:
Does this mean 100 us? Maybe worth a comment.

Reply:
yes, I will add a comment here.